core/terraform/appgateway/appgateway.tf

```diff
@@ -224,4 +224,11 @@ resource "azurerm_monitor_diagnostic_setting" "agw" {
   lifecycle { ignore_changes = [log_analytics_destination_type] }
 }
 
+# Once the deployment of the app gateway is complete, we can proceed to include the required DNS zone for Nexus, which is dependent on the FQDN of the app gateway.
+resource "azurerm_private_dns_zone" "non_core" {
```

Comment:
1. The name of the resource isn't right
2. Non core dns zones have their own file. I think this should move there.
3. Have you tested the system starting with a version before this change and an installed nexus, then do the upgrade to the new version including this change? I think it might fail and needs attention as to how we do the migration.

Reply:
1. Changed the name to "nexus"
2. Moved to the right location
3. Fixed and tested in the migrartion.sh